src/JWT/Payload.php

```
@@ -0,0 +1,230 @@
+    {
+        return $this->payload['iat'];
+    }
+
+    /**
+     * Get the not before claim
+     *
+     * @return mixed|null
+     */
+    public function nbf()
+    {
+        return $this->payload['nbf'];
+    }
+
+    /**
+     * Get the token id claim
+     *
+     * @return mixed|null
+     */
+    public function jti()
+    {
+        return $this->payload['jti'];
+    }
+
+    /**
+     * Get the JWT payload
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        return $this->payload;
+    }
+
+    /**
+     * Get the payload as an array
+     *
+     * @param integer $options
+     * @return string
+     */
+    public function toJson($options = 0)
+    {
+        return json_encode($this->payload, $options);
+    }
+
+    /**
+     * Get the payload as an array
+     *
+     * @return string
+     */
+    public function jsonSerialize()
+    {
+        return $this->payload;
+    }
+
+    /**
+     * Get the payload as JSON
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->toJson();
+    }
+
+    /**
+     * Dynamically check a claim exist
+     *
+     * @param string $claim
+     * @return bool
+     */
+    public function __isset(string $claim)
+    {
+        return $this->offsetExists($claim);
+    }
+
+    /**
+     * Dynamically get a claim
+     *
+     * @param string $claim
+     * @return mixed
+     */
+    public function __get(string $claim)
+    {
+        return $this->offsetGet($claim);
+    }
+
+    /**
+     * Dynamically set a claim
+     *
+     * @param string $claim
+     * @param mixed $value
+     * @return void
+     */
+    public function __set(string $claim, $value)
+    {
+        $this->offsetSet($claim, $value);
+    }
+
+    /**
+     * Dynamically unset a claim
+     *
+     * @param string $claim
+     * @return void
+     */
+    public function __unset(string $claim)
+    {
+        $this->offsetUnset($claim);
+    }
```

An exception should be throw if trying to set/unset properties on the payload - it should be readonly.